Add recipe for diffsync

ber-ro opened the pull request:
### Brief summary of what the package does

Diffsync uses recursive diff output as a base for directory synchronization.
There are shortcuts for the following operations: copy, move, delete, ediff, find-file.

### Direct link to the package repository

https://github.com/ber-ro/diffsync

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm with `x`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this, and thanks for your close adherence to the checklist.  A couple clarifying questions about this -

- How does this compare to a unix cli tool like rsync?
- Is this intended to be a windows-only package?  I ask since I see it calls "diff.exe" as in `diffsync-revert-buffer`; if that's the case it may be worth mentioning that this is a win32-specific package

ber-ro commented:
Added a screenshot to Readme.

- I was unsure how to name it. My first idea was 'diffdir', because for me it is a replacement for ediff-directories. But as I use it for synchronization/merging of directories I thought it would be a good idea to mention the 'sync' aspect. Like ediff the purpose is an interactive usage. So maybe the 'sync' part of the name is a bit misleading.
- I removed the EXE suffix. It is also not needed for msys2 or cygwin.

riscy commented:
Thanks!  Looks like an interesting package.  One quick comment as I'm merging is that `(message "%s" "Diffsync running diff...done")` and similar forms can be replaced by just `(message "Diffsync running diff...done")`.
